clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -301,20 +302,33 @@ object GarbageCollector {
   }
 
   def main(args: Array[String]) {
-    if (args.length != 2) {
+    if (args.length < 1) {
       Console.err.println(
-        "Usage: ... <repo_name> <region>"
+        "Usage: ... <repo_name> [region]"
       )
       System.exit(1)
     }
 
     val spark = SparkSession.builder().appName("GarbageCollector").getOrCreate()
+    val hc = spark.sparkContext.hadoopConfiguration
+
+    val apiURL = hc.get(LAKEFS_CONF_API_URL_KEY)
+    val accessKey = hc.get(LAKEFS_CONF_API_ACCESS_KEY_KEY)
+    val secretKey = hc.get(LAKEFS_CONF_API_SECRET_KEY_KEY)
+    val apiClient = new ApiClient(apiURL, accessKey, secretKey)
+    val storageType = apiClient.getBlockstoreType()
+
+    if (storageType == StorageTypeS3 && args.length != 2) {
+      Console.err.println(
+        "Usage: ... <repo_name> <region>"
+      )
+      System.exit(1)
+    }
 
     val repo = args(0)
-    val region = args(1)
+    val region = if (args.length == 2) args(1) else ""
```

Comment:
```suggestion
    val region = if (args.length == 2) args(1) else null
```
`null` is a perfectly good `String` on the JVM, and it is more clearly a "missing value" than the empty string `""`.